dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx

```diff
@@ -274,23 +274,47 @@ export const DataConfigPanelItem = ({ fieldOptionList, visualizations }: any) =>
     </>
   );
 
+  const reduceFields = (dataFields: IField[], dataType: string) => {
+    return dataFields.reduce((acc: IField[], field: IField) => {
+      let isFilterable: boolean;
+      switch (dataType) {
+        case 'geo_point':
+          isFilterable = field.type === dataType;
+          break;
+        case 'number':
+          isFilterable = field.type === 'float' || field.type === 'integer';
+          break;
+        default:
+          isFilterable = true;
+      }
+      if (isFilterable) {
+        acc.push({ ...field, label: field.name });
+        return acc;
```

Comment:
I feel this `return acc;` is duplicated here at 292 as later it would return acc anyway, it only needs to push one item and that's it in this if block.